conftest.py

```diff
@@ -92,14 +92,11 @@ def pytest_collection_modifyitems(config, items):
     shard_index = int(os.environ["TVM_SHARD_INDEX"])
 
     print(f"Marking tests for shard {shard_index} of {num_shards}")
-    for item in items:
+    items_copy = list(items)
+    for item in items_copy:
         item_shard_index = find_shard_index(item.nodeid, num_shards=num_shards)
-        item.add_marker(
-            pytest.mark.skipif(
-                item_shard_index != shard_index,
-                reason=f"Test running on shard {item_shard_index} of {num_shards}",
-            )
-        )
+        if item_shard_index != shard_index:
+            items.remove(item)
```

Comment:
The only hesitation I have with this is that now if someone goes and looks for a test named `test_abc` in one of the 10 CPU integration shard logs, they'd have to search through every log until they find it and more importantly know that it's not weird that the  test doesn't show up at all in most of the logs. Do you have any ideas on how we could make this a better ux?

Reply:
I am not very sure how we could make this a better UX. 
I understand your concern, but I'm wondering how often someone does a manual search for a specific test through the shard logs. If a test fails, the corresponding shard will appear red in the UI. 

Alternatively, we can keep the old log structure (showing the skipped tests), and I can filter them later when processing the XML logs.

Reply:
I see what you mean, I suppose it would be pretty rare. In a follow up or something we can add a message that links to the log where all the steps are catted together